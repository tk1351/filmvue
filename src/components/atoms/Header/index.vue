<script setup lang="ts">
import TextLink from '../TextLink/index.vue'

const links = [
  { title: 'AAA', href: '#' },
  { title: 'BBB', href: '#' },
  { title: 'CCC', href: '#' },
]

interface Emits {
  (eventName: 'click:titleLink', event: Event): void
}
const emit = defineEmits<Emits>()

const handleTitleLink = (event: Event) => {
  event.preventDefault()
  emit('click:titleLink', event)
}
</script>

<template>
  <header class="header">
    <h1 class="header__title">
      <a data-test-id="title-link" @click="handleTitleLink"> FilmVue </a>
    </h1>
    <nav>
      <ul class="header__list">
        <li v-for="link in links" :key="link.title">
          <TextLink :href="link.href" :title="link.title" color="secondary" />
        </li>
      </ul>
    </nav>
  </header>
</template>

<style scoped lang="scss">
.header {
  height: 70px;
  padding: 0 40px;
  background-color: $main-color;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.header__title {
  &:hover {
    cursor: pointer;
  }
}

.header__list {
  display: flex;
  gap: 50px;
}
</style>
