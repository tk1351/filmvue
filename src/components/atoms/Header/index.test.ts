import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import Header from './index.vue'

describe('Testing Header Component', () => {
  it('正しくHeaderコンポーネントが表示されていること', () => {
    const wrapper = mount(Header)
    expect(wrapper.element).toMatchSnapshot()
  })

  describe('Event', () => {
    it('Titleをクリックすると、 click:titleLink イベントが1回実行されること', async () => {
      const wrapper = mount(Header)
      await wrapper.find('[data-test-id="title-link"]').trigger('click')
      expect(wrapper.emitted('click:titleLink')).toHaveLength(1)
    })
  })
})
