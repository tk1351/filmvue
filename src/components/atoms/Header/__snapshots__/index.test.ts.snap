// Vitest Snapshot v1

exports[`Testing Header Component > 正しくHeaderコンポーネントが表示されていること 1`] = `
<header
  class="header"
  data-v-e8a0fba6=""
>
  <h1
    class="header__title"
    data-v-e8a0fba6=""
  >
    <a
      data-test-id="title-link"
      data-v-e8a0fba6=""
    >
       FilmVue 
    </a>
  </h1>
  <nav
    data-v-e8a0fba6=""
  >
    <ul
      class="header__list"
      data-v-e8a0fba6=""
    >
      
      <li
        data-v-e8a0fba6=""
      >
        <a
          class="text-link -secondary"
          data-v-8475e6b2=""
          data-v-e8a0fba6=""
          href="#"
        >
          AAA
        </a>
      </li>
      <li
        data-v-e8a0fba6=""
      >
        <a
          class="text-link -secondary"
          data-v-8475e6b2=""
          data-v-e8a0fba6=""
          href="#"
        >
          BBB
        </a>
      </li>
      <li
        data-v-e8a0fba6=""
      >
        <a
          class="text-link -secondary"
          data-v-8475e6b2=""
          data-v-e8a0fba6=""
          href="#"
        >
          CCC
        </a>
      </li>
      
    </ul>
  </nav>
</header>
`;
